drivers/entropy/entropy_renesas_ra.c

```diff
@@ -35,15 +33,22 @@ static int entropy_ra_rsip_trng_get_entropy(const struct device *dev, uint8_t *b
 	return 0;
 }
 
-static DEVICE_API(entropy, entropy_ra_rsip_trng_api) = {
-	.get_entropy = entropy_ra_rsip_trng_get_entropy,
+static DEVICE_API(entropy, entropy_renesas_ra_api) = {
+	.get_entropy = entropy_renesas_ra_get_entropy,
 };
 
-static int entropy_ra_rsip_trng_init(const struct device *dev)
+static int entropy_renesas_ra_init(const struct device *dev)
 {
 	HW_SCE_McuSpecificInit();
 	return 0;
 }
 
-DEVICE_DT_INST_DEFINE(0, entropy_ra_rsip_trng_init, NULL, NULL, NULL, PRE_KERNEL_1,
-		      CONFIG_ENTROPY_INIT_PRIORITY, &entropy_ra_rsip_trng_api);
+#if DT_NODE_HAS_STATUS_OKAY(DT_INST(0, renesas_ra_rsip_e51a_trng))
```

Comment:
In this case, using `DT_FOREACH_STATUS_OKAY` is the proper way.
This code does not assume that multiple instances exist, but this implementation should be satisfactory if it does.

https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/display/ssd1306.c#L530-L531

Reply:
- I had tried to use DT_FOREACH_STATUS_OKAY, but it cannot work, because the trng node doesn't have a full path reference (with address append the node name). So, using this macro cause a build error.
- In actual use, there is no case that have 2 TRNG node. You can check HWM of RA6M1, RA6M2, RA6M3 (SCE7) and RA8M1, RA8D1, RA8T1 (RSIP), there is only 1 unit of security module (SCE and RSIP)

Reply:
Can't it be done like this?
I haven't tried all of them, but RA6M3 and RA8M1 seem to compile correctly.

```
diff --git a/drivers/entropy/entropy_renesas_ra.c b/drivers/entropy/entropy_renesas_ra.c
index fb81671de41..87980a51863 100644
--- a/drivers/entropy/entropy_renesas_ra.c
+++ b/drivers/entropy/entropy_renesas_ra.c
@@ -43,12 +43,9 @@ static int entropy_renesas_ra_init(const struct device *dev)
        return 0;
 }

-#if DT_NODE_HAS_STATUS_OKAY(DT_INST(0, renesas_ra_rsip_e51a_trng))
-DEVICE_DT_DEFINE(DT_INST(0, renesas_ra_rsip_e51a_trng), entropy_renesas_ra_init, NULL, NULL, NULL,
-                PRE_KERNEL_1, CONFIG_ENTROPY_INIT_PRIORITY, &entropy_renesas_ra_api)
-#endif
-
-#if DT_NODE_HAS_STATUS_OKAY(DT_INST(0, renesas_ra_sce7_rng))
-DEVICE_DT_DEFINE(DT_INST(0, renesas_ra_sce7_rng), entropy_renesas_ra_init, NULL, NULL, NULL,
-                PRE_KERNEL_1, CONFIG_ENTROPY_INIT_PRIORITY, &entropy_renesas_ra_api)
-#endif
+#define RENESAS_RA_ENTROPY_INIT(node) \
+       DEVICE_DT_DEFINE(node, entropy_renesas_ra_init, NULL, NULL, NULL, \
+                        PRE_KERNEL_1, CONFIG_ENTROPY_INIT_PRIORITY, &entropy_renesas_ra_api)
+
+DT_FOREACH_STATUS_OKAY(renesas_ra_rsip_e51a_trng, RENESAS_RA_ENTROPY_INIT)
+DT_FOREACH_STATUS_OKAY(renesas_ra_sce7_rng, RENESAS_RA_ENTROPY_INIT)
```

Reply:
I updated for it. Thanks for your suggestion!